include/gatekeeper_gk.h

```diff
@@ -24,13 +24,94 @@
 #include "gatekeeper_ipip.h"
 #include "gatekeeper_ggu.h"
 #include "gatekeeper_mailbox.h"
+#include "gatekeeper_lpm.h"
+
+/* The maximum number of gateways that Gatekeeper may use. */
+#define GK_MAX_NUM_GW       (4)
+
+/*
+ * The LPM supports 8-bit next hop,
+ * which at most has 128 different values.
+ */
+#define GK_MAX_NUM_POLICIES (128)
 
 /*
  * A flow entry can be in one of three states:
  * request, granted, or declined.
  */
 enum gk_flow_state { GK_REQUEST, GK_GRANTED, GK_DECLINED };
 
+enum gk_policy_action {
+	/* Forward the packet to a Grantor server. */
+	GK_FWD_GT,
+
+	/* Forward the packet to the back interface. */
+	GK_FWD_BACK_NET,
+
+	/* Drop the packet. */
+	GK_DROP,
+
+	/* Invalid policy action. */
+	GK_POLICY_MAX,
+};
+
+struct gk_gateway {
+
+	uint16_t proto;
+
+	union {
+		struct in_addr  v4;
+		struct in6_addr v6;
+	} ip;
+
+	bool is_valid;
+};
+
+struct gk_policy {
+
+	/* The policy action. */
+	enum gk_policy_action   action;
+
+	/*
+	 * Although unlikely, a Grantor server could be
+	 * in the back network of a Gatekeeper server,
+	 * no gateway would be needed in this case.
+	 */
+	bool                    is_gt_in_back_net;
+
+	/*
+	 * The gateway for the grantor when
+	 * the Grantor server is not in the back network.
+	 */
+	struct gk_gateway       *gw;
```

Comment:
Do we need to avoid the second lookup of the LPM table by having a reference from the grantor entry to the gateway entry? Or we also need to resolve the Grantor server through the LPM table to find the gateway in a second lookup?